packages/api/tsserver/tsserver.mts

```diff
@@ -0,0 +1,134 @@
+import { parseTsServerMessages } from './utils.mjs';
+import type { ChildProcess } from 'child_process';
+import type { server as tsserver } from 'typescript';
+
+export class TsServer {
+  private _seq: number = 0;
+  private readonly process: ChildProcess;
+  private readonly resolvers: Record<number, (value: any) => void> = {};
+
+  constructor(process: ChildProcess) {
+    this.process = process;
+    this.process.stdout?.on('data', (buffer) => {
+      for (const message of parseTsServerMessages(buffer)) {
+        if (message.type === 'response') {
+          this.handleResponse(message);
```

Comment:
Can we add a bit of documentation on what responses an d events here from the TSServer's point of view?

Reply:
The types are defined in protocol.ts in the typescript code base and we define the exact types we expect in this file already. That's good enough for what we need.